backend/test_observer/data_access/repository.py

```diff
@@ -130,6 +139,35 @@ def get_artefacts_by_family(
     return query.all()
 
 
+def get_historic_test_executions(
+    session: Session,
+    test_execution: TestExecution,
+) -> list[TestExecution]:
```

Comment:
In my mind a repository is more of a simple wrapper that provides some helper methods to access the database. Perhaps business logic should not reside in it. I would move this function to [here](https://github.com/canonical/test_observer/blob/main/backend/test_observer/controllers/test_executions/logic.py)